pkg/datapath/tables/node_address.go

```diff
@@ -265,24 +272,43 @@ func (n *nodeAddressController) register() {
 
 }
 
+func (n *nodeAddressController) updateK8sNodeIPs(node node.LocalNode) (updated bool) {
+	if ip := node.GetNodeIP(true); ip != nil {
+		if newIP, ok := netip.AddrFromSlice(ip); ok {
+			if newIP != n.k8sIPv6 {
+				n.k8sIPv6 = newIP
+				updated = true
+			}
+		}
+	}
+	if ip := node.GetNodeIP(false); ip != nil {
+		if newIP, ok := netip.AddrFromSlice(ip); ok {
+			if newIP != n.k8sIPv4 {
+				n.k8sIPv4 = newIP
+				updated = true
+			}
+		}
+	}
+	return
+}
+
 func (n *nodeAddressController) run(ctx context.Context, reporter cell.Health) error {
 	defer n.deviceChanges.Close()
 
+	localNodeChanges := stream.ToChannel(ctx, n.LocalNode)
+	n.updateK8sNodeIPs(<-localNodeChanges)
+
 	limiter := rate.NewLimiter(nodeAddressControllerMinInterval, 1)
 	for {
 		txn := n.DB.WriteTxn(n.NodeAddresses)
 		for change, _, ok := n.deviceChanges.Next(); ok; change, _, ok = n.deviceChanges.Next() {
 			dev := change.Object
 
-			// Note: prefix match! existing may contain node addresses from devices with names
```

Comment:
Note to self/backporter: Verify whether this code is needed in v1.15 or whether this workaround is still needed.